lib/s3-s3.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * AWS.S3-like behavior with failover to a secondary S3 location
+ */
+
+'use strict';
+
+// Used in init() below, this function mimics AWS.S3
+var s3 = function (primaryS3, secondaryS3) {
```

Comment:
Well I do think this would look slightly nicer and am changing to it:

```
s3 = new S3S3(new AWS.S3(awsConfigPrimary), new AWS.S3(awsConfigSecondary));
```

But yes, I did want to make sure that aws-sdk was not pulled in.  And by having a proxy of everything, it's clearer what is really supported and what really doesn't try to failover (since the latter right now throws an error rather than silently only working with just the primary).